File: 프로그래머스/2/389479. 서버 증설 횟수/서버 증설 횟수.py
def solution(players, m, k):
    server = [0] * 24
    answer = 0
    
    for i in range(0, 24):
        need_server = (players[i] - m) // m + 1 if players[i] >= m else 0
        server_count = 0
        
        for j in range(max(0, i - k + 1), i):
            server_count += server[j]
        
        server[i] = max(0, need_server - server_count)
        answer += server[i]
    
    return answer

File: 프로그래머스/2/389479. 서버 증설 횟수/서버 증설 횟수.java
class Solution {
    public int solution(int[] players, int m, int k) {
        int[] server = new int[24];
        int answer = 0;
        
        for (int i = 0; i < 24; i++) {
            int needServer = 0;
            if (players[i] >= m) {
                needServer += ((players[i] - m) / m) + 1;
            }
            
            int serverCnt = 0;
            for (int j = Math.max(0, i - k + 1); j < i; j++) {
                serverCnt += server[j];
            }
            
            int add = Math.max(0, needServer - serverCnt);
            server[i] = add;
            answer += add;
        }
        
        return answer;
    }
}